### ISSUE TITLE
Easy-RSA error:  signing failed

### ISSUE THREAD

**Issue by lucaszanella** [kylemanna/docker-openvpn#437]
```
writing new private key to '/etc/openvpn/pki/private/cilinho.dyndns.org.key.XXXXhMhCkM'
-----
Using configuration from /usr/share/easy-rsa/openssl-easyrsa.cnf
Enter pass phrase for /etc/openvpn/pki/private/ca.key:
unable to load CA private key
1995662544:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:531:
1995662544:error:23077074:PKCS12 routines:PKCS12_pbe_crypt:pkcs12 cipherfinal error:p12_decr.c:108:
1995662544:error:2306A075:PKCS12 routines:PKCS12_item_decrypt_d2i:pkcs12 pbe crypt error:p12_decr.c:139:
1995662544:error:0907B00D:PEM routines:PEM_READ_BIO_PRIVATEKEY:ASN1 lib:pem_pkey.c:142:

Easy-RSA error:

signing failed (openssl output above may have more detail)


```

I'm pretty sure I'm using the same password I've inputted before. What could be happening?

**Comment by amateusz** [comment]
Same here. I cannot generate client certs anymore

**Comment by dortonway** [comment]
The same :(

**Comment by kkleidal** [comment]
I was having the same issue, even when rebuilding from master like @amateusz suggested, but then I tried using a stronger password and it worked. Maybe there is some password-entropy requirement that fails with a bad error message. More likely and EasyRSA problem than a problem with this repo.

*tldr: use a stronger password*

**Comment by deepsidhu1313** [comment]
Edited Dockerfile and replaced 
```
FROM alpine:latest 
```

with

```
FROM alpine:3.8
```
And worked. I guess it is unstable docker image file.

**Comment by stevevanhoyweghen** [comment]
I have the same issue.
...
e is 65537 (0x010001)
Can't load /etc/openvpn/pki/.rnd into RNG
140618083146600:error:2406F079:random number generator:RAND_load_file:Cannot open file:crypto/rand/randfile.c:98:Filename=ls.rnd
...

Tried several password from complex to simple. No success.

easy-rsa issue https://github.com/OpenVPN/easy-rsa/issues/261 seems related and is fixed now. Is this (one of the) root causes?

**Comment by utamav** [comment]
I also have the same issue. Stronger password does not seem to help. Any fix or suggestions?

**Comment by J216** [comment]
I am also also having the same issue. It worked a month ago, now I can't sign request, though I can generate them. I was just starting to feel like I had easyRSA and openVPN figured out, now it doesn't work.
...
Enter pass phrase for /home/amber/EasyRSA-3.0.4/pki/private/ca.key:
unable to load CA private key
139714059077264:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:544:
139714059077264:error:23077074:PKCS12 routines:PKCS12_pbe_crypt:pkcs12 cipherfinal error:p12_decr.c:104:
139714059077264:error:2306A075:PKCS12 routines:PKCS12_item_decrypt_d2i:pkcs12 pbe crypt error:p12_decr.c:130:
139714059077264:error:0907B00D:PEM routines:PEM_READ_BIO_PRIVATEKEY:ASN1 lib:pem_pkey.c:132:
...

**Comment by casenjo** [comment]
Doesn't seem like the latest alpine image (3.10.1 as of this post) has fixed this issue. The solution that @deepsidhu1313 suggested is what's working for me too

**Comment by MountainLynx** [comment]
I also encountered same issue on a headless server, and the problem for me at least, seems to have been that the entropy on my device was rather low (around 700, checked by _ cat /proc/sys/kernel/random/entropy_avail_), because running haveged generator immediately solved the problem. Is it possible low entropy has something to do with this issue?

**Comment by maurerle** [comment]
I had this issue too.
I read for example [here](https://security.stackexchange.com/a/95352) that smashing your keyboard while generating dh parameters would speed up this process. This is bad in this case, as characters typed while generating dh params in the same shell are not lost and are instead part of the passphrase inserted afterwards, which makes the passphrase invalid.

So make sure to type the same amount of backspace if you typed on your keyboard while generating dh params

To create entropy you can instead use a different shell on the server.

Maybe the usability could be improved here or mention this in the README/docs.

**Comment by maurerle** [comment]
this issue is a duplicate of #323 and #148

**Comment by Dieterm5** [comment]
I have the same issue :(((

```
Using SSL: openssl OpenSSL 1.1.1g  21 Apr 2020
Generating a RSA private key
.+++++
...........................................+++++
writing new private key to '/etc/openvpn/pki/easy-rsa-1.MANEBF/tmp.ImFLdD'
-----
Using configuration from /etc/openvpn/pki/easy-rsa-1.MANEBF/tmp.eNAJap
Enter pass phrase for /etc/openvpn/pki/private/ca.key:
User interface error
139850348707144:error:2807106B:UI routines:UI_process:processing error:crypto/ui/ui_lib.c:545:while reading strings
unable to load CA private key
139850348707144:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:crypto/evp/evp_enc.c:583:
139850348707144:error:0906A065:PEM routines:PEM_do_header:bad decrypt:crypto/pem/pem_lib.c:461:

Easy-RSA error:

signing failed (openssl output above may have more detail)

Easy-RSA error:

Failed to sign 'user'
```

I removed the image and downloaded it again, but it's not working.
I want to run this image on synology NAS, anyone could help me?

Thanks
